calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -260,14 +262,26 @@ def _generate_rules(self):
         inbound_rules = [
             {
                 'action': 'allow',
-            },
+            }
         ]
 
         outbound_rules = [
             {
                 'action': 'allow',
-            },
+            }
         ]
+
+        print('Getting Policy Rules from Annotation of pod %s' % pod)
+
+        annotations = self._get_metadata(pod, 'annotations')
+        namespace = self._get_metadata(pod, 'namespace')
+
+        if 'allowFrom' in annotations.keys():
+            inbound_rules = load_json(annotations['allowFrom'])
```

Comment:
calico has some pretty strict rule requirements. It will complain if you try to add an invalid rule.